### PR TITLE
Fix iteration_space

### PR DESCRIPTION
**Pull request by jdahm**
## Technical Description

Fix my original changes to the parsing for the iteration_space, and add tests to catch problems next time.

### CODE REVIEW

**Comment by jdahm** [comment]
launch jenkins

File: gtclang/test/unit-test/Frontend/TestPreprocessing.cpp [hunk 1]
```diff
@@ -29,24 +32,86 @@ namespace {
...
+    // iteration_space(i_start, i_end - 1)
+    dawn::ast::VerticalRegionDeclStmt* vrds =
+        dawn::dyn_cast<dawn::sir::VerticalRegionDeclStmt>(statements[0].get());
+    ASSERT_NE(vrds, nullptr);
+    checkVerticalRegion(*vrds->getVerticalRegion(), Interval(LevelKind::Start, LevelKind::End),
+                        std::optional<Interval>(Interval(LevelKind::Start, LevelKind::End, 0, -1)),
+                        std::optional<Interval>());
+  }
+
+  std::cout << "2" << std::endl;
```
Comment (eddie-c-davis) [inline gtclang/test/unit-test/Frontend/TestPreprocessing.cpp]: Are these trace statements left intentionally?

File: gtclang/test/unit-test/Frontend/TestPreprocessing.cpp [hunk 2]
```diff
@@ -14,7 +14,10 @@
 //
 //===------------------------------------------------------------------------------------------===//
 
+// #include "dawn/AST/ASTVisitor.h"
```
Comment (eddie-c-davis) [inline gtclang/test/unit-test/Frontend/TestPreprocessing.cpp]: Leftover commented include

File: gtclang/src/gtclang/Frontend/GTClangPreprocessorAction.cpp [hunk 3]
```diff
@@ -313,26 +313,50 @@ class GTClangLexer {
...
           const std::array<char, 3> coordChar{'i', 'j', 'k'};
-          for(char e : coordChar) {
-            const std::string& boundStart = *boundIter;
-            ++boundIter;
-            const std::string& boundEnd = *boundIter;
-            const std::string errMsg = std::string{"failed parsing iteration_space argument: "} +
-                                       "(" + boundStart + "," + boundEnd + ")";
-            if(e != boundStart[0]) {
+
+          auto boundIter = std::begin(curBounds);
```
Comment (eddie-c-davis) [inline gtclang/src/gtclang/Frontend/GTClangPreprocessorAction.cpp]: Definitely much cleaner, nicely done

File: gtclang/src/gtclang/Frontend/GTClangPreprocessorAction.cpp [hunk 4]
```diff
@@ -877,9 +901,9 @@ class GTClangLexer {
 
   /// @brief Get the replacements
   const std::vector<clang::tooling::Replacement>& getReplacements() const { return replacements_; }
-};
+}; // namespace
```
Comment (twicki) [inline gtclang/src/gtclang/Frontend/GTClangPreprocessorAction.cpp]: this is a leftover from a parenthesis mismatch

File: gtclang/test/unit-test/Frontend/TestPreprocessing.cpp (hunk 1 above)
Comment (jdahm) [reply gtclang/test/unit-test/Frontend/TestPreprocessing.cpp]: Thanks for catching!

File: gtclang/src/gtclang/Frontend/GTClangPreprocessorAction.cpp (hunk 3 above)
Comment (jdahm) [reply gtclang/src/gtclang/Frontend/GTClangPreprocessorAction.cpp]: 😄

File: gtclang/src/gtclang/Frontend/GTClangPreprocessorAction.cpp (hunk 4 above)
Comment (jdahm) [reply gtclang/src/gtclang/Frontend/GTClangPreprocessorAction.cpp]: Thanks for catching!

**Comment by jdahm** [comment]
launch jenkins

**Pull request merged**